src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SemanticModelExtensions.cs

```diff
@@ -362,44 +362,48 @@ public static string GenerateNameForExpression(
             current = current.WalkDownParentheses();
 
             if (current is IdentifierNameSyntax identifierName)
-            {
                 return identifierName.Identifier.ValueText.ToCamelCase();
-            }
-            else if (current is MemberAccessExpressionSyntax memberAccess)
-            {
+
+            if (current is MemberAccessExpressionSyntax memberAccess)
                 return memberAccess.Name.Identifier.ValueText.ToCamelCase();
-            }
-            else if (current is MemberBindingExpressionSyntax memberBinding)
-            {
+
+            if (current is MemberBindingExpressionSyntax memberBinding)
                 return memberBinding.Name.Identifier.ValueText.ToCamelCase();
-            }
-            else if (current is ConditionalAccessExpressionSyntax conditionalAccess)
-            {
-                current = conditionalAccess.WhenNotNull;
-            }
-            else if (current is CastExpressionSyntax castExpression)
-            {
-                current = castExpression.Expression;
-            }
-            else if (current is DeclarationExpressionSyntax decl)
+
+            if (current is DeclarationExpressionSyntax decl)
             {
                 if (decl.Designation is not SingleVariableDesignationSyntax name)
-                {
                     break;
-                }
 
                 return name.Identifier.ValueText.ToCamelCase();
             }
-            else if (current.Parent is ForEachStatementSyntax foreachStatement &&
-                     foreachStatement.Expression == expression)
+
+            if (current.Parent is ForEachStatementSyntax foreachStatement &&
+                foreachStatement.Expression == expression)
             {
                 var word = foreachStatement.Identifier.ValueText.ToCamelCase();
                 return CodeAnalysis.Shared.Extensions.SemanticModelExtensions.Pluralize(word);
             }
-            else
+
+            if (current.Parent is AnonymousObjectMemberDeclaratorSyntax { NameEquals: { } nameEquals } anonymousObjectMemberDeclarator &&
+                anonymousObjectMemberDeclarator.Expression == current)
             {
-                break;
+                return nameEquals.Name.Identifier.ValueText.ToCamelCase();
```

Comment:
>nameEquals.Name.Identifier.ValueText.ToCamelCase(); [](http://example.com/codeflow?start=23&length=51)

There isn't an option for determining how to case this?

Reply:
Probably.  But this just follows what we already do in this method.  And it doesn't look like we have customer complaining about locals naming (looks like it's pretty universal).